app/src/main/java/us/shandian/giga/get/DownloadManagerImpl.java

```diff
@@ -23,7 +35,8 @@
     private static final String TAG = DownloadManagerImpl.class.getSimpleName();
     private final DownloadDataSource mDownloadDataSource;
 
-    private final ArrayList<DownloadMission> mMissions = new ArrayList<DownloadMission>();
+    private final ArrayList<DownloadMission> mMissions = new ArrayList<>();
+    private final Context context;
```

Comment:
This should at least be labelled as `NonNull`. 

Ideally, it's better to pass in a listener/runnable with a function to call when an exception occurs in this case, just to avoid keeping `Context` in a class unless there's no better solution.